added macos dependency installation

rrmanukyan opened the pull request:
Added missing dependency installation on generic MacOS runner


## High Level Overview of Change


Installing Python and cmake before the build process


### Type of Change

<!--
Please check [x] relevant options, delete irrelevant ones.
-->

- [x] Chore (no impact to binary, e.g. `.gitignore`, formatting, dropping support for older tooling)

## Test Plan
A test pipeline was run on the forked repo

shichengripple001 reviewed (commented):
I think these can be moved to the github-runner provisioning pipeline, so it wont be run for every build

rrmanukyan commented:
> I think these can be moved to the github-runner provisioning pipeline, so it wont be run for every build

When dependencies are installed, each step takes around a second and the packages are not being reinstalled every time. 
example: https://github.com/ripple/rippledavid/actions/runs/12603322049/job/35128142629

The reason it was not put in runner provisioning script is to keep it generic and the build pipeline runner agnostic.

shichengripple001 commented:
we may need to specify the python version at least, once there is new python version, it might  potentially cause some issue.
Also I remember brew install will try to upgrade versions for install packages, that might take some time.

rrmanukyan commented:
> we may need to specify the python version at least, once there is new python version, it might potentially cause some issue. Also I remember brew install will try to upgrade versions for install packages, that might take some time.

Added Python version and moved both python and cmake installation under condition which will prevent unexpected updates.
